plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -376,7 +381,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
-        return getColumns(icebergTable.schema(), typeManager).stream()
+        ImmutableList.Builder<IcebergColumnHandle> columns = ImmutableList.builder();
```

Comment:
Can you avoid building the interim List and go straight to building the final Map?

Reply:
You could structure it as:

```
ImmutableMap.builder columnHandles = ...
for (IcebergColumnHandle columnHandle : getColumns(...) {
    columnHandles.put(...)
}
columnHandles.put(FILE_PATH.name(), pathColumnHandle());
return columnHandles.build();
```

Reply:
updated @alexjo2144